core/src/main/java/ai/timefold/solver/core/config/util/ConfigUtils.java

```diff
@@ -481,16 +458,7 @@ public static boolean isGenericTypeImmutable(Class<?> type) {
         if (type == null) {
             return false;
         }
-        return BULT_IN_IMMUTABLE_CLASSES.contains(type);
-    }
-
-    /**
-     * @param memberAccessor never null
-     * @return true if the field is integer; otherwise false
-     */
-    public static boolean isIntegerType(MemberAccessor memberAccessor) {
-        var field = (Class<?>) memberAccessor.getGenericType();
-        return field.isAssignableFrom(Integer.class) || field.isAssignableFrom(int.class);
+        return IMMUTABLE_CLASSES.contains(type) || type.isRecord();
```

Comment:
For performance, flip the two. One will do the hash, the other is just a boolean.